### ISSUE TITLE
Deprecate aws_ses_domain_identity_verification - SES now uses DKIM to verify domains

### ISSUE THREAD

**Issue by OJFord** [hashicorp/terraform-provider-aws#28286]
### Description

While it appears to still be in the API for now, and thus not actually broken yet, in the console the domain verification page shows a deprecation warning/CSV download only for 'legacy' TXT verification records:
![image](https://user-images.githubusercontent.com/3946014/206902497-c6f90d2d-2cb6-47f5-ad60-42e86e7ca93e.png)

The documentation page appears to make no mention of these legacy records, only discussing verification by DKIM:
https://docs.aws.amazon.com/ses/latest/dg/creating-identities.html#verify-domain-procedure

I believe the ` aws_ses_domain_identity_verification` should be documented as deprecated, with a pointer to `aws_ses_domain_dkim` as the new way of achieving the same (it was previously possible to verify a domain by TXT record, and use it without DKIM).

I haven't been able to find any documentation pages explicitly deprecating the old method, or mentioning if/when it will stop working  (or if it has actually? I haven't tested it - I have _existing_ domains verified that way, which the resource is able to refresh for, but I haven't tested actually _creating_ the resource/verifying that way) for new domains.

### Would you like to implement a fix?

None

**Comment by jpriebe** [comment]
@OJFord - I think you might be on to something here.  It looks like the provider is using this call:

https://docs.aws.amazon.com/sdk-for-go/api/service/ses/#SES.GetIdentityVerificationAttributesWithContext

for `aws_ses_domain_identity_verification`.  Docs for that call indicate it is using the TXT record method of verification:

> For domain identities, the domain's verification status is "Pending" as Amazon SES searches for the required TXT record in the DNS settings of the domain. When Amazon SES detects the record, the domain's verification status changes to "Success". If Amazon SES is unable to detect the record within 72 hours, the domain's verification status changes to "Failed." In that case, if you still want to verify the domain, you must restart the verification process from the beginning.

Question: can you initiate the verification step from terraform if you're using `aws_ses_domain_dkim` instead of `aws_ses_domain_identity_verification` ?

**Comment by Dunedan** [comment]
> Question: can you initiate the verification step from terraform if you're using aws_ses_domain_dkim instead of aws_ses_domain_identity_verification ?

Yes, using `aws_ses_domain_dkim` and `aws_route53_record` for the DKIM CNAME records is sufficient nowadays. AWS does the verification now based on the DKIM CNAME records instead of the TXT record.

**Comment by OJFord** [comment]
Yes, there's nothing to initiate though it just happens.

For at least a period if not always, both were possible, i.e. you could _not_ use DKIM and use the verification TXT record instead, or you could use DKIM and not need it.

The most recent change is that you _must_ use DKIM, and thus there's no need for a separate record to verify domain ownership ever.

**Comment by kamilturek** [comment]
I can confirm that verifying domain identities with a TXT record still works fine. However, as already pointed out in this thread, DKIM is currently the recommended method. I believe we could update the documentation page with a note similar to this one:

<img width="1037" alt="image" src="https://github.com/hashicorp/terraform-provider-aws/assets/22010517/e11ca24d-bf4f-47b8-a0b9-22058df3cecb">

**Comment by OJFord** [comment]
This remains an issue and has an open PR linked above.
